dynacred/src/config.ts

```diff
@@ -0,0 +1,76 @@
+import * as toml from "toml";
+
+/**
+    * Type containing both the addresses of the endpoints for servers and for
+    * clients
+    */
+class Addresses {
+    constructor(
+        readonly servers: string[],
+        readonly clients: string[],
+    ) {} // tslint:disable-line
+}
+
+/**
+ * Readonly container for the configuration
+ *
+ * TODO preciser name
+ */
+export class Config {
+    constructor(
+        readonly addresses: Addresses,
+    ) {}
+}
+
+/**
+ * Load the Config from the server
+ */
+export default async function LoadConfig(): Promise<Config> {
+    const res = await fetch(window.location.origin + "/assets/conodes.toml");
+    if (!res.ok) {
+        return Promise.reject(new Error(`on fetch config: ${res.status}: ${res.body}`));
+    }
+
+    const parsed = toml.parse(await res.text());
+    if (!("servers" in parsed)) {
+        return Promise.reject(new Error(`on parse config: unable to find "servers"`));
+    }
+    if (!(Symbol.iterator in parsed.servers)) {
+        return Promise.reject(new Error(`on parse config: "servers" is not iterable`));
+    }
+
+    const addrsClient: string[] = [];
+    const addrsServer: string[] = [];
+    for (const s of parsed.servers) {
+        const getField = (key: string): Promise<string> => {
+            if (!(key in s)) {
+                return Promise.reject(new Error(`on parse config: unable to find "${key}" in each server`));
+            }
+
+            const value = s[key];
+            if (typeof value !== "string") {
+                return Promise.reject(new Error(`on parse config: "${key}" of server is not a string`));
+            }
+            if (!value.startsWith("tls:")) {
+                return Promise.reject(new Error(`on parse config: "${key}" of server doesn't start with "tls:"`));
+            }
+
+            return Promise.resolve(value);
+        };
+
+        const addr = await getField("Address");
+        const url = await getField("Url").catch((err) => {
```

Comment:
I think the field-name is `URL`, all-caps.

Reply:
It seems to be `Url` (from https://github.com/dedis/cothority/blob/master/dedis-cothority.toml)

Reply:
Oh no - it seems different version do different things :( Here is what I have on my server:

```
$ cat conode_data/public.toml 
[[servers]]
  Address = "tls://conode.c4dt.org:7770"
  URL = "https://conode.c4dt.org:7771"
  Suite = "Ed25519"
...
```

So it should accept both `Url` and `URL`.

Reply:
Found the following in the latest java-code:

```
                // Special case: allow URL when it should be Url
                if (srvToml.contains("URL")) {
                    srv.Url = srvToml.getString("URL");
                }
```

Reply:
So, we have a choice now, do we support ill-formatted configuration or not (I lean towards the latter), if not, which format do we take?

Reply:
After discussion with @ineiti: support both, correct format it `Url`.

Reply:
Done.